### PR TITLE
t2 crash-reporter: adds argv (sans first two entries)

### PR DESCRIPTION
**Pull request by rwaldron**
@tikurahul what do you think about this? There will be a new `argv` property that gets posted to the crash report server. We can try it out for a bit before I write a bunch of tests.

Signed-off-by: Rick Waldron <waldron.rick@gmail.com>

### CODE REVIEW

**Comment by tikurahul** [comment]
@rwaldron This looks good to me. I can add support for `argv` in the server side. I will add it to the core model and also make it searchable in the search index. I will mark `argv` parameter as optional in the REST API.

**Comment by rwaldron** [comment]
Sounds like a plan!

File: lib/crash-reporter.js [hunk 1]
```diff
@@ -142,14 +146,18 @@ CrashReporter.submit = function(report, opts) {
...
+
   return new Promise((resolve, reject) => {
     request.post({
-      url: SUBMIT_CRASH_URL,
+      url,
       form: {
-        crash: report,
-        labels: labels,
-        f: 'json'
+        argv,
```
Comment (tikurahul) [inline lib/crash-reporter.js]: Can you do an `argv.join(',')` to make it consistent with labels.

File: lib/crash-reporter.js [hunk 2]
```diff
@@ -107,6 +107,10 @@ CrashReporter.submit = function(report, opts) {
   }
 
   const silent = (opts.silent || false);
+  // Removes:
+  // - The path to node
+  // - The path to t2
+  const argv = process.argv.slice(2).join(', ');
```
Comment (rwaldron) [inline lib/crash-reporter.js]: @tikurahul updated here

File: test/unit/crash-reporter.js [hunk 3]
```diff
@@ -206,6 +206,28 @@ exports['CrashReporter.submit'] = {
...
+    }).then(() => {
+      test.equal(this.crPost.callCount, 1);
+      test.equal(this.crPost.lastCall.args[2], 'foo');
+      process.argv[2] = original;
+      test.done();
+    }).catch(error => {
+      test.ok(false, error.message);
+      test.done();
+    });
+  },
```
Comment (rwaldron) [inline test/unit/crash-reporter.js]: @HipsterBrown @johnnyman727 test added.

**Comment by tikurahul** [comment]
:+1: LGTM

**Comment by rwaldron** [comment]
@tikurahul thanks!

**Pull request merged**